src/gui/XournalppCursor.cpp

```diff
@@ -302,16 +325,32 @@ GdkCursor* XournalppCursor::getEraserCursor()
 GdkCursor* XournalppCursor::getHighlighterCursor()
 {
 	XOJ_CHECK_TYPE(XournalppCursor);
+	
+	if (this->drawDirActive)
+	{
+		this->drawDirActive = false;
```

Comment:
This is error prone because it modifies the state: updating the cursor twice a row will not create the correct cursor.